### PR TITLE
Only keep traceback reprs, to avoid cycles

### PR DESCRIPTION
**Pull request by bryevdv**
@manopapad is this what you had in mind? With this change on your cycle-detector PR, there is still a cycle reported, but it is considerably less output than is reported without this change. Spcifically, the remaining cycle does not go through `traceback` like the ones without this change, so perhaps this should be merged as a discrete improvement in that area specifically. 

### With this PR:

```
test38 ❯ LEGATE_PRECISE_EXCEPTION_TRACE=1 legate /tmp/a.py -legate:cycle-check
looking for cycles involving 0x7f17b1c36af0, of type <class 'legate.core.store.RegionField'>
looking for cycles involving 0x7f17b1c42100, of type <class 'legate.core.store.RegionField'>
looking for cycles involving 0x7f17b1c36e20, of type <class 'legate.core.store.RegionField'>
looking for cycles involving 0x7f17b1c42970, of type <class 'legate.core._legion.future.Future'>
found cycle!
  tail:
    0x7f17b1c42970: <class 'legate.core._legion.future.Future'>
     ^ ["_future"]
    0x7f17b292d800: <class 'dict'>
     ^ .__dict__
    0x7f17b1c36f40: <class 'legate.core.exception.PendingException'>
     ^ [0]
    0x7f17b2933600: <class 'list'>
     ^ ["_pending_exceptions"]
    0x7f17b2914800: <class 'dict'>
     ^ .__dict__
    0x7f17b2b55e80: <class 'legate.core.runtime.Runtime'>
     ^ ["legate_runtime"]
    0x7f17b1b1bf40: <class 'dict'>
  cycle:
    0x7f17b1b1bf40: <class 'dict'>
     ^ .__dict__
    0x7f17b1b13d90: <class 'cunumeric.runtime.Runtime'>
     ^ ["runtime"]
    0x7f17c190cc80: <class 'dict'>
     ^ .__dict__
    0x7f17c19038b0: <class 'cunumeric.config.CuNumericLib'>
     ^ ["_library"]
    0x7f18d0d6a9c0: <class 'dict'>
     ^ .__dict__
    0x7f17b2b24ca0: <class 'legate.core.context.Context'>
     ^ ["legate_context"]
    0x7f17b1b1bf40: <class 'dict'>
```

### Without this PR

<details>

```
test38 ❯ LEGATE_PRECISE_EXCEPTION_TRACE=1 legate /tmp/a.py -legate:cycle-check
looking for cycles involving 0x7fe6e45f1a30, of type <class 'legate.core.store.RegionField'>
found cycle!
  tail:
    0x7fe6e45f1a30: <class 'legate.core.store.RegionField'>
     ^ ["_data"]
    0x7fe7ecad90c0: <class 'dict'>
     ^ .__dict__
    0x7fe6e44f5640: <class 'legate.core.store.Storage'>
     ^ ["_storage"]
    0x7fe800765280: <class 'dict'>
     ^ .__dict__
    0x7fe6e44f5610: <class 'legate.core.store.Store'>
     ^ ["base"]
    0x7fe800330b80: <class 'dict'>
     ^ .__dict__
    0x7fe6e44f57f0: <class 'cunumeric.deferred.DeferredArray'>
     ^ ["_thunk"]
    0x7fe8002cd340: <class 'dict'>
     ^ .__dict__
    0x7fe6e44efbe0: <class 'cunumeric.array.ndarray'>
     ^ [0]
    0x7fe6e4648f40: <class 'tuple'>
     ^ [1]
    0x7fe6df054680: <class 'tuple'>
     ^
    0x7fe6df02cc10: <class 'frame'>
  cycle:
    0x7fe6df02cc10: <class 'frame'>
     ^ .tb_frame
    0x7fe6e4476800: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476b00: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476680: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476880: <class 'traceback'>
     ^ ["_tb"]
    0x7fe6e4487180: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1670: <class 'legate.core.exception.PendingException'>
     ^ [0]
    0x7fe6e4487240: <class 'list'>
     ^ ["_pending_exceptions"]
    0x7fe7ecc0d040: <class 'dict'>
     ^ .__dict__
    0x7fe6e47365b0: <class 'legate.core.runtime.Runtime'>
     ^ ["legate_runtime"]
    0x7fe6df224a80: <class 'dict'>
     ^ .__dict__
    0x7fe6dee57dc0: <class 'cunumeric.runtime.Runtime'>
     ^ ["runtime"]
    0x7fe6e465fe80: <class 'dict'>
     ^ .f_globals
    0x7fe6df02cc10: <class 'frame'>
looking for cycles involving 0x7fe6e45fb040, of type <class 'legate.core.store.RegionField'>
found cycle!
  tail:
    0x7fe6e45fb040: <class 'legate.core.store.RegionField'>
     ^ ["_data"]
    0x7fe80033aa00: <class 'dict'>
     ^ .__dict__
    0x7fe6e44f5a00: <class 'legate.core.store.Storage'>
     ^ ["_storage"]
    0x7fe8002acfc0: <class 'dict'>
     ^ .__dict__
    0x7fe6e44f5a60: <class 'legate.core.store.Store'>
     ^
    0x7fe6c4c94cf0: <class 'frame'>
  cycle:
    0x7fe6c4c94cf0: <class 'frame'>
     ^ .tb_frame
    0x7fe6e47ce380: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476140: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476900: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476400: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476740: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476800: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476b00: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476680: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476880: <class 'traceback'>
     ^ ["_tb"]
    0x7fe6e460ddc0: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1ee0: <class 'legate.core.operation.AutoTask'>
     ^
    0x7fe6c4c94cf0: <class 'frame'>
looking for cycles involving 0x7fe6e45f1d60, of type <class 'legate.core.store.RegionField'>
found cycle!
  tail:
    0x7fe6e45f1d60: <class 'legate.core.store.RegionField'>
     ^ ["_data"]
    0x7fe800306380: <class 'dict'>
     ^ .__dict__
    0x7fe6e44e4b80: <class 'legate.core.store.Storage'>
     ^ ["_storage"]
    0x7fe800306a00: <class 'dict'>
     ^ .__dict__
    0x7fe6e44e4b20: <class 'legate.core.store.Store'>
     ^ ["base"]
    0x7fe8002b97c0: <class 'dict'>
     ^ .__dict__
    0x7fe7ecac6220: <class 'cunumeric.deferred.DeferredArray'>
     ^
    0x7fe6e45ff640: <class 'frame'>
  cycle:
    0x7fe6e45ff640: <class 'frame'>
     ^ .tb_frame
    0x7fe6e4476900: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476400: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476740: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476800: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476b00: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476680: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476880: <class 'traceback'>
     ^ ["_tb"]
    0x7fe6e4487180: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1670: <class 'legate.core.exception.PendingException'>
     ^ [0]
    0x7fe6e4487240: <class 'list'>
     ^ ["_pending_exceptions"]
    0x7fe7ecc0d040: <class 'dict'>
     ^ .__dict__
    0x7fe6e47365b0: <class 'legate.core.runtime.Runtime'>
     ^ ["legate_runtime"]
    0x7fe6df224a80: <class 'dict'>
     ^ .__dict__
    0x7fe6dee57dc0: <class 'cunumeric.runtime.Runtime'>
     ^ ["runtime"]
    0x7fe6e465f800: <class 'dict'>
     ^ .f_globals
    0x7fe6e460e200: <class 'frame'>
     ^ .f_back
    0x7fe6e45ff640: <class 'frame'>
looking for cycles involving 0x7fe6e45fb490, of type <class 'legate.core.store.RegionField'>
found cycle!
  tail:
    0x7fe6e45fb490: <class 'legate.core.store.RegionField'>
     ^ ["_data"]
    0x7fe6e44c4cc0: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1c70: <class 'legate.core.store.Storage'>
     ^ ["_storage"]
    0x7fe8002ac140: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f17c0: <class 'legate.core.store.Store'>
     ^ ["base"]
    0x7fe6e44c4740: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1be0: <class 'cunumeric.deferred.DeferredArray'>
     ^
    0x7fe6e45ff640: <class 'frame'>
  cycle:
    0x7fe6e45ff640: <class 'frame'>
     ^ .tb_frame
    0x7fe6e4476900: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476400: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476740: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476800: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476b00: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476680: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476880: <class 'traceback'>
     ^ ["_tb"]
    0x7fe6e4487180: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1670: <class 'legate.core.exception.PendingException'>
     ^ [0]
    0x7fe6e4487240: <class 'list'>
     ^ ["_pending_exceptions"]
    0x7fe7ecc0d040: <class 'dict'>
     ^ .__dict__
    0x7fe6e47365b0: <class 'legate.core.runtime.Runtime'>
     ^ ["legate_runtime"]
    0x7fe6df224a80: <class 'dict'>
     ^ .__dict__
    0x7fe6dee57dc0: <class 'cunumeric.runtime.Runtime'>
     ^ ["runtime"]
    0x7fe6e465f800: <class 'dict'>
     ^ .f_globals
    0x7fe6e460e200: <class 'frame'>
     ^ .f_back
    0x7fe6e45ff640: <class 'frame'>
looking for cycles involving 0x7fe6e45fb8b0, of type <class 'legate.core._legion.future.Future'>
found cycle!
  tail:
    0x7fe6e45fb8b0: <class 'legate.core._legion.future.Future'>
     ^ ["_future"]
    0x7fe6e4487180: <class 'dict'>
  cycle:
    0x7fe6e4487180: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1670: <class 'legate.core.exception.PendingException'>
     ^ [0]
    0x7fe6e4487240: <class 'list'>
     ^ ["_pending_exceptions"]
    0x7fe7ecc0d040: <class 'dict'>
     ^ .__dict__
    0x7fe6e47365b0: <class 'legate.core.runtime.Runtime'>
     ^ ["legate_runtime"]
    0x7fe6df224a80: <class 'dict'>
     ^ .__dict__
    0x7fe6dee57dc0: <class 'cunumeric.runtime.Runtime'>
     ^ ["runtime"]
    0x7fe6e465fe80: <class 'dict'>
     ^ .f_globals
    0x7fe6df02cc10: <class 'frame'>
     ^ .tb_frame
    0x7fe6e4476800: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476b00: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476680: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476880: <class 'traceback'>
     ^ ["_tb"]
    0x7fe6e4487180: <class 'dict'>
looking for cycles involving 0x7fe6e45fb8e0, of type <class 'legate.core.store.RegionField'>
found cycle!
  tail:
    0x7fe6e45fb8e0: <class 'legate.core.store.RegionField'>
     ^ ["_data"]
    0x7fe80036a280: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1df0: <class 'legate.core.store.Storage'>
     ^ ["_storage"]
    0x7fe8003340c0: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f14c0: <class 'legate.core.store.Store'>
     ^ ["base"]
    0x7fe8002b9340: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f16a0: <class 'cunumeric.deferred.DeferredArray'>
     ^
    0x7fe6e45ff640: <class 'frame'>
  cycle:
    0x7fe6e45ff640: <class 'frame'>
     ^ .tb_frame
    0x7fe6e4476900: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476400: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476740: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476800: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476b00: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476680: <class 'traceback'>
     ^ .tb_next
    0x7fe6e4476880: <class 'traceback'>
     ^ ["_tb"]
    0x7fe6e4487180: <class 'dict'>
     ^ .__dict__
    0x7fe6e45f1670: <class 'legate.core.exception.PendingException'>
     ^ [0]
    0x7fe6e4487240: <class 'list'>
     ^ ["_pending_exceptions"]
    0x7fe7ecc0d040: <class 'dict'>
     ^ .__dict__
    0x7fe6e47365b0: <class 'legate.core.runtime.Runtime'>
     ^ ["legate_runtime"]
    0x7fe6df224a80: <class 'dict'>
     ^ .__dict__
    0x7fe6dee57dc0: <class 'cunumeric.runtime.Runtime'>
     ^ ["runtime"]
    0x7fe6e465f800: <class 'dict'>
     ^ .f_globals
    0x7fe6e460e200: <class 'frame'>
     ^ .f_back
    0x7fe6e45ff640: <class 'frame'>
```

</details>

### CODE REVIEW

File: legate/core/operation.py [hunk 1]
```diff
@@ -279,7 +277,7 @@ def can_raise_exception(self) -> bool:
         return len(self._exn_types) > 0
 
     def capture_traceback(self) -> None:
-        self._tb = capture_traceback()
+        self._tb = capture_traceback_repr()
```
Comment (manopapad) [inline legate/core/operation.py:280]: `self._tb_repr`

**Comment by manopapad** [comment]
With this change we seem to be losing the stacktrace corresponding to the original location of the faulty operation.

Try running this example:

```
import cunumeric as cn
a = cn.arange(10)
b = cn.arange(9)
c = cn.ones((9,))
d = c[a]
e = c[b]
```

as follows:

```
LEGATE_MAX_PENDING_EXCEPTIONS=2 LEGATE_PRECISE_EXCEPTION_TRACE=1 LEGATE_TEST=1 legate c.py -cunumeric:test
```

The selection of env vars means we don't block immediately to check that potentially failing tasks return successfully, but instead we can "run ahead" for up to 2 operations (`LEGATE_MAX_PENDING_EXCEPTIONS=2`) before blocking to check on them, but we still want to know exactly which operation raised the exception (`LEGATE_PRECISE_EXCEPTION_TRACE=1`), which is why we're caching their pending stacktraces (the alternative would be to just report the exception at the point where we actually blocked, which allows us to skip recording stacktraces, but would result in an imprecise error message).

<details>
<summary>Before</summary>

```
Traceback (most recent call last):
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "c.py", line 5, in <module>
    d = c[a]
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()
IndexError: index is out of bounds in index array

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "c.py", line 6, in <module>
    e = c[b]
  File "/Users/mpapadakis/legate.core/legate/core/context.py", line 241, in wrapper
    result = func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()
  File "/Users/mpapadakis/legate.core/legate/core/operation.py", line 183, in execute
    self._context.runtime.submit(self)
  File "/Users/mpapadakis/legate.core/legate/core/runtime.py", line 1185, in submit
    self.raise_exceptions()
  File "/Users/mpapadakis/legate.core/legate/core/runtime.py", line 1575, in raise_exceptions
    pending.raise_exception()
  File "/Users/mpapadakis/legate.core/legate/core/exception.py", line 49, in raise_exception
    raise exn_reraised from exn_original
IndexError: index is out of bounds in index array
```

</details>




<details>
<summary>After</summary>

```
IndexError: index is out of bounds in index array

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "c.py", line 6, in <module>
    e = c[b]
  File "/Users/mpapadakis/legate.core/legate/core/context.py", line 241, in wrapper
    result = func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()
  File "/Users/mpapadakis/legate.core/legate/core/operation.py", line 181, in execute
    self._context.runtime.submit(self)
  File "/Users/mpapadakis/legate.core/legate/core/runtime.py", line 1183, in submit
    self.raise_exceptions()
  File "/Users/mpapadakis/legate.core/legate/core/runtime.py", line 1573, in raise_exceptions
    pending.raise_exception()
  File "/Users/mpapadakis/legate.core/legate/core/exception.py", line 49, in raise_exception
    raise exn_reraised from exn_original
IndexError: index is out of bounds in index array
```

</details>

File: legate/core/operation.py (hunk 1 above)
Comment (bryevdv) [reply legate/core/operation.py:280]: not sure why mypy isn't flagging this tbh

Comment (manopapad) [reply legate/core/operation.py:280]: maybe we need to explicitly enable such a check?

**Comment by bryevdv** [comment]
@manopapad right, need to actually use `format_tb` explicitly. This is the output with 72a14ab does that look good?
```
dev39 ❯ LEGATE_MAX_PENDING_EXCEPTIONS=2 LEGATE_PRECISE_EXCEPTION_TRACE=1 LEGATE_TEST=1 legate bt.py -cunumeric:test
[0 - 7f4a1cd60000]    0.919895 {6}{python}: python exception occurred within task:
IndexError:   File "/home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "bt.py", line 5, in <module>
    d = c[a]
  File "/home/bryan/work/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()

index is out of bounds in index array

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "bt.py", line 6, in <module>
    e = c[b]
  File "/home/bryan/work/legate.core/legate/core/context.py", line 241, in wrapper
    result = func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()
  File "/home/bryan/work/legate.core/legate/core/operation.py", line 181, in execute
    self._context.runtime.submit(self)
  File "/home/bryan/work/legate.core/legate/core/runtime.py", line 1171, in submit
    self.raise_exceptions()
  File "/home/bryan/work/legate.core/legate/core/runtime.py", line 1561, in raise_exceptions
    pending.raise_exception()
  File "/home/bryan/work/legate.core/legate/core/exception.py", line 49, in raise_exception
    raise exn_reraised from exn_original
IndexError:   File "/home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "bt.py", line 5, in <module>
    d = c[a]
  File "/home/bryan/work/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/home/bryan/work/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/home/bryan/work/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()

index is out of bounds in index array
legion_python: /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-src/runtime/realm/python/python_module.cc:996: virtual void Realm::LocalPythonProcessor::execute_task(Realm::Processor::TaskFuncID, const Realm::ByteArrayRef&): Assertion `0' failed.
Signal 6 received by node 0, process 102506 (thread 7f4a1cd60000) - obtaining backtrace
Signal 6 received by process 102506 (thread 7f4a1cd60000) at: stack trace: 14 frames
  [0] = /lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7f4a25bb2090]
  [1] = /lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb) [0x7f4a25bb200b]
  [2] = /lib/x86_64-linux-gnu/libc.so.6(abort+0x12b) [0x7f4a25b91859]
  [3] = /lib/x86_64-linux-gnu/libc.so.6(+0x22729) [0x7f4a25b91729]
  [4] = /lib/x86_64-linux-gnu/libc.so.6(+0x33fd6) [0x7f4a25ba2fd6]
  [5] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x562ccc) [0x7f4a264afccc]
  [6] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x505a92) [0x7f4a26452a92]
  [7] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x505b26) [0x7f4a26452b26]
  [8] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x56685a) [0x7f4a264b385a]
  [9] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x509470) [0x7f4a26456470]
  [10] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x565547) [0x7f4a264b2547]
  [11] = /home/bryan/work/legate.core/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib/librealm.so.1(+0x50d35f) [0x7f4a2645a35f]
  [12] = /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7f4a24425609]
  [13] = /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7f4a25c8e133]
```

**Comment by manopapad** [comment]
@bryevdv I made some minor changes, to keep the exception name together with the rest of the message, and clean up the "reraised" exception stacktrace. I think this is good, let me know if you agree.

```
IndexError: index is out of bounds in index array
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "c.py", line 5, in <module>
    d = c[a]
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 446, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "c.py", line 6, in <module>
    e = c[b]
  File "/Users/mpapadakis/legate.core/legate/core/context.py", line 241, in wrapper
    result = func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 144, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 936, in __getitem__
    return ndarray(shape=None, thunk=self._thunk.get_item(key))
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 828, in get_item
    ) = self._create_indexing_array(key)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 734, in _create_indexing_array
    output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 484, in _zip_indices
    task.execute()
  File "/Users/mpapadakis/legate.core/legate/core/operation.py", line 181, in execute
    self._context.runtime.submit(self)
  File "/Users/mpapadakis/legate.core/legate/core/runtime.py", line 1171, in submit
    self.raise_exceptions()
  File "/Users/mpapadakis/legate.core/legate/core/runtime.py", line 1561, in raise_exceptions
    pending.raise_exception()
  File "/Users/mpapadakis/legate.core/legate/core/exception.py", line 53, in raise_exception
    raise exn_reraised from exn_original
IndexError: index is out of bounds in index array
```

**Comment by bryevdv** [comment]
Oh that's close to what I had at first, but I changed it based on the output above. Definitely what ever output you think is most clear for users!

**Pull request merged**